ouroboros-consensus-protocol/ouroboros-consensus-protocol.cabal

```diff
@@ -63,3 +63,20 @@ library
   if flag(asserts)
     ghc-options:
       -fno-ignore-asserts
+
+library ouroboros-consensus-protocol-test
```

Comment:
I think we have avoided private libraries so far. Would it be too disproportionate to continue the pattern of a fresh `ouroboros-consensus-protocol-test` directory and `ouroboros-consensus-protocol-test.cabal` file?

Reply:
But we avoided private libraries because various tools didn't support them (notably, haskell.nix). Now it does!

Reply:
Clarke has pointed out that upstream libraries (eg Plutus and Ledger) use this. So if support for it breaks us then it will have broken for them and presumably they'll be the ones engaging with `haskell.nix` team eg to fix it

Reply:
For future reference:

- https://cabal.readthedocs.io/en/3.6/cabal-package.html?highlight=libraries#sublibs  ("internal libraries" were the first way to have multiple libraries per package)

- https://fgaz.me/posts/2019-11-14-cabal-multiple-libraries/ (then Gazzetta's GSoC let those multiple libraries also be public)

Reply:
Remaining reservation is that this breaks our packaging scheme. So my TODO is: do we want to migrate towards this, and so this is the first step towards using the new feature?

Reply:
If you wouldn't mind: `src: test-src`